moto/awslambda/models.py

```diff
@@ -429,24 +431,39 @@ def _create_zipfile_from_plaintext_code(code):
 class EventSourceMapping(BaseModel):
     def __init__(self, spec):
         # required
-        self.function_name = spec['FunctionName']
+        self.function_arn = spec['FunctionArn']
         self.event_source_arn = spec['EventSourceArn']
-        self.starting_position = spec['StartingPosition']
-
+        self.uuid = str(uuid.uuid4())
+        self.last_modified = time.mktime(datetime.datetime.utcnow().timetuple())
         # optional
-        self.batch_size = spec.get('BatchSize', 100)
+        self.starting_position = spec.get('StartingPosition', 'TRIM_HORIZON')
+        self.batch_size = spec.get('BatchSize', 10)  # TODO: Add source type-specific defaults
```

Comment:
BatchSize is currently set to the following https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html:

> Amazon Kinesis - Default 100. Max 10,000.
Amazon DynamoDB Streams - Default 100. Max 1,000.
Amazon Simple Queue Service - Default 10. Max 10.

Otherwise, this field should be blank (add a check if it's Kinesis, DynamoDB, or SQS)